### PR TITLE
EZP-30627: Unified Twig helpers naming

### PR DESCRIPTION
**Pull request by pawbuj**
| Question           | Answer
| ------------------ | ------------------
| **JIRA issue**     | [EZP-30627](https://jira.ez.no/browse/EZP-30627)
| **Bug/Improvement**| yes
| **New feature**    | no
| **Target version** |`master`
| **BC breaks**      | yes
| **Tests pass**     | yes
| **Doc needed**     | yes

related to : https://github.com/ezsystems/ezplatform-admin-ui/pull/1008

**TODO**:
- [ ] Implement feature / fix a bug.
- [ ] Implement tests.
- [ ] Fix new code according to Coding Standards (`$ composer fix-cs`).
- [ ] Ask for Code Review.

### CODE REVIEW

**Comment by gggeek** [comment]
Following up the discussion at https://github.com/ezsystems/ezpublish-kernel/pull/2633 : were the current names of twig helpers already declared as deprecated? If not, I'd politely suggest to keep around the old name (and add a deprecation notice for any usage) for at least the duration of one major kernel version. Twig helpers tend to be one of the apis most used by every ez developer...

**Comment by emodric** [comment]
I fully agree with @gggeek and his sentiment expressed in #2663. And is also valid for #2658 .

File: eZ/Publish/Core/MVC/Symfony/Templating/Twig/Extension/ContentExtension.php [hunk 1]
```diff
@@ -208,44 +204,6 @@ public function getTranslatedFieldDefinitionDescription(ValueObject $content, $f
...
-     *
-     * @param \eZ\Publish\API\Repository\Values\ValueObject $object Can be any kid of Value object which directly holds the translated data
-     * @param string $property Property name, example 'name', 'description'
-     * @param string $forcedLanguage Locale we want the content name translation in (e.g. "fre-FR"). Null by default (takes current locale)
-     *
-     * @throws \eZ\Publish\Core\Base\Exceptions\InvalidArgumentValue If $property does not exists as plural or as method
-     *
-     * @return string|null
-     */
-    public function getTranslatedProperty(ValueObject $object, $property, $forcedLanguage = null)
```
Comment (alongosz) [inline eZ/Publish/Core/MVC/Symfony/Templating/Twig/Extension/ContentExtension.php]: Why did you drop it? Seems out of scope.

File: eZ/Publish/Core/MVC/Symfony/Templating/Twig/Extension/CoreExtension.php [hunk 2]
```diff
@@ -39,6 +39,6 @@ public function getName()
      */
     public function getGlobals()
     {
-        return array('ezpublish' => $this->globalHelper);
+        return ['ez_app' => $this->globalHelper];
```
Comment (alongosz) [inline eZ/Publish/Core/MVC/Symfony/Templating/Twig/Extension/CoreExtension.php]: Rather `ezplatform`, it's about the Product.

File: eZ/Publish/Core/MVC/Symfony/Templating/Twig/Extension/RichTextExtension.php [hunk 3]
```diff
@@ -39,20 +39,20 @@ public function getFilters()
     {
         return array(
             new Twig_SimpleFilter(
-                'richtext_to_html5',
+                'ez_richtext_to_html5',
```
Comment (alongosz) [inline eZ/Publish/Core/MVC/Symfony/Templating/Twig/Extension/RichTextExtension.php]: No point in changing this, it's being dropped via #2659

File: eZ/Bundle/EzPublishCoreBundle/Resources/views/content_fields.html.twig [hunk 4]
```diff
@@ -27,7 +27,7 @@
 {% endblock %}
 
 {% block ezrichtext_field %}
-    {%- set field_value = field.value.xml|richtext_to_html5 -%}
+    {%- set field_value = field.value.xml|ez_richtext_to_html5 -%}
```
Comment (alongosz) [inline eZ/Bundle/EzPublishCoreBundle/Resources/views/content_fields.html.twig]: Same here

**Comment by lserwatka** [comment]
Due to large amount of changes for Symfony 4 and cleanup around templates directory structures, this change does not add any significant work which needs to be done during upgrade anyway, especially that we are considering to provide a simple bash/PHP command or entire compatibly bundle. Subject is still open.

**Comment by alongosz** [comment]
@pawbuj besides my previous review please list those changes in `/doc/bc/changes-8.0.md` so it's easy to find them for anyone upgrading.

File: eZ/Publish/Core/MVC/Symfony/Templating/Twig/Extension/ContentExtension.php (hunk 1 above)
Comment (pawbuj) [reply eZ/Publish/Core/MVC/Symfony/Templating/Twig/Extension/ContentExtension.php]: because I dropped  `ez_trans_prop`

File: eZ/Publish/Core/MVC/Symfony/Templating/Twig/Extension/CoreExtension.php (hunk 2 above)
Comment (pawbuj) [reply eZ/Publish/Core/MVC/Symfony/Templating/Twig/Extension/CoreExtension.php]: Symfony provides `app` global helper , ezPlatform provides `ez_app` . So, it is because of naming compatibility with symfony.

File: eZ/Publish/Core/MVC/Symfony/Templating/Twig/Extension/RichTextExtension.php (hunk 3 above)
Comment (pawbuj) [reply eZ/Publish/Core/MVC/Symfony/Templating/Twig/Extension/RichTextExtension.php]: #2659 PR is still open, I can not taking it in account.

File: eZ/Publish/Core/MVC/Symfony/Templating/Twig/Extension/CoreExtension.php (hunk 2 above)
Comment (alongosz) [reply eZ/Publish/Core/MVC/Symfony/Templating/Twig/Extension/CoreExtension.php]: Still -1 from my side

File: eZ/Publish/Core/MVC/Symfony/Templating/Twig/Extension/ContentExtension.php (hunk 1 above)
Comment (alongosz) [reply eZ/Publish/Core/MVC/Symfony/Templating/Twig/Extension/ContentExtension.php]: > because I dropped `ez_trans_prop`

Not following, scope was to unify naming, not drop features. Is there a replacement for this one?

File: eZ/Publish/Core/MVC/Symfony/Templating/Twig/Extension/CoreExtension.php (hunk 2 above)
Comment (Nattfarinn) [reply eZ/Publish/Core/MVC/Symfony/Templating/Twig/Extension/CoreExtension.php]: In most cases (with exceptions of course, but this is not the case here) renaming `ezpublish` to `ezplatform` is recommended.

**Comment by MarioBlazek** [comment]
@lserwatka does it mean that the next version of eZ Platform is going to be 100% Symfony 4/5 compliant?
